### PR TITLE
#159054017 Follow/Unfollow profiles feature

### PR DESCRIPTION
**Pull request by asamisellah**
#### What does this PR do?
Allows users to follow other users and unfollow them. It also enables user to get a list of people they follow and people following them.

#### Description of Task to be completed?
- Write tests for the feature
- Modify profile model to include a follows field with a many-to-many relationship with the model
- Develop methods to follow and unfollow profiles
- Include urls for the views
- Develop views for the follow/unfollow function and get followers and following profiles

##### Endpoints

METHOD | ENDPOINT | FUNCTIONALITY 
------------ | -------------|--------------
POST | /api/profiles/:username/follow/ | Follows a profile
DELETE | /api/profiles/:username/follow/ | Unfollows a profile
GET | /api/profiles/:username/followers/ | Gets list of followers
GET | /api/profiles/:username/following/ | Gets list of following

#### How should this be manually tested?
- Clone this repo

```
cd ah-jarvis

git pull origin ft-follow-unfollow-authors-159054017
  
create and activate a virtual env 

pip install -r requirements.txt

python manage.py migrate

coverage run manage.py test

```
- When all the tests pass, it means that the feature works. 

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#159054017](https://www.pivotaltracker.com/story/show/159054017/comments/192845380) Users should be able to follow each other

### CODE REVIEW

File: authors/apps/profiles/tests/test_follow_profile.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+from django.utils.encoding import force_bytes
+from ...authentication.utils import generate_token
+from rest_framework import status
+
+
+class TestFollowAPI(APITestCase):
+    ''' Tests the followAPI view. '''
+
+    def test_user_can_follow_profile(self):
+        ''' Tests that a user can follow another user '''
```
Comment (dwarukira) [inline authors/apps/profiles/tests/test_follow_profile.py:17]: Use double quotes doc string

**Comment by denniswanjiru** [comment]
You should change the endpoints to have a single slash instead of 2

**Comment by asamisellah** [comment]
@DennisWanjiru I've made the correction.

**Review by dwarukira — CHANGES_REQUESTED** [review]

**Comment by Duncanian** [comment]
@asamisellah  , add a token to the following methods in `authors.apps.authentication.tests.test_profile.TestUserProfile`

- test_get_profile 
- test_non_existent_user

It will solve your error as the profile needs an authenticated user.

**Pull request merged**